src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -669,9 +674,23 @@ public class DeleteOldRevisionsResult
             }
 
             var deleted = DeleteRevisionsInternal(context, table, lowerIdPrefix, collectionName, changeVector, lastModifiedTicks, result.PreviousCount, revisionsToDelete, result);
-
             IncrementCountOfRevisions(context, lowerIdPrefix, -deleted);
             result.Remaining = result.PreviousCount - deleted;
+
+            if (conflicted &&
+                nonPersistentFlags.Contain(NonPersistentDocumentFlags.ByEnforceRevisionConfiguration)==false &&
+                nonPersistentFlags.Contain(NonPersistentDocumentFlags.FromSmuggler)==false &&
+                deleted > 0)
+            {
+                var lowerId = lowerIdPrefix.ToString();
+                lowerId = lowerId.Substring(0, lowerId.Length - 1); // cut the prefix last char (SpecialChars.RecordSeparator)
```

Comment:
perform this operation on notification write

Reply:
The slice cannot be overwritten when you dispose the context?

if so, you cannot pass the slice and extract the id from it when the notification is written, because the context can be disposed untill then..